### PR TITLE
ui: restore --delete indicates number of deleted files

### LINKED ISSUE #5092: `restic restore --delete` output should indicate number of files deleted
Output of `restic version`
--------------------------

027
What should restic do differently? Which functionality do you think we should add?
----------------------------------------------------------------------------------
Restic restore --delete output should indicate number of files deleted

output indicates number of files restore/skipped but does not indicate number of files deleted
This is helpful to know the command worked

What are you trying to do? What problem would this solve?
---------------------------------------------------------

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------

### PR DESCRIPTION
**Pull request by konidev20**
What does this PR change? What problem does it solve?
-----------------------------------------------------
* adds new field `FilesDeleted` to the State struct, JSON and text progress updaters
* renamed `ReportDeletedFile` to `ReportDeletion`
* increment FilesDeleted count when ReportedDeletion is called

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------
Closes #5092 

Checklist
---------
- [x] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [x] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I have added tests for all code changes.
- [x] I have added documentation for relevant changes (in the manual).
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- [x] I have run `gofmt` on the code in all commits.
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
- [x] I'm done! This pull request is ready for review.

### CODE REVIEW

File: internal/ui/restore/progress.go [hunk 1]
```diff
@@ -129,6 +130,8 @@ func (p *Progress) ReportDeletedFile(name string) {
 		return
 	}
 
+	p.s.FilesDeleted++
```
Comment (MichaelEischer) [inline internal/ui/restore/progress.go:133]: Files are deleted using `fs.RemoveAll` which means that a single name can actually refer to a full folder of files. It feels wrong to count it as a single item in that case.

Comment (konidev20) [reply internal/ui/restore/progress.go:133]: @MichaelEischer, I see. I checked the logic, I have modified this place to do a file traversal to count the files, run the `fs.RemoveAll`, and then finally update the count of removed files. Is there something I might still be missing?

File: internal/restorer/restorer.go [hunk 2]
```diff
@@ -511,12 +511,32 @@ func (res *Restorer) removeUnexpectedFiles(ctx context.Context, target, location
...
 		// only delete files that were selected for restore
 		if selectedForRestore {
-			res.opts.Progress.ReportDeletedFile(nodeLocation)
+			// First collect all files that will be deleted
+			var filesToDelete []string
+			err := filepath.Walk(nodeTarget, func(path string, info os.FileInfo, err error) error {
+				if err != nil {
+					return err
+				}
+				if !info.IsDir() {
```
Comment (MichaelEischer) [inline internal/restorer/restorer.go:520]: `ReportDeletedFile` previously also reported deletions of directories. I think we should keep that behavior. That is, just collect all directory entries that would be deleted. And ideally rename `ReportDeletedFile` to `ReportDeletion`.

**Review by MichaelEischer — COMMENTED** [review]
I have only one small nit remaining for the code. Besides that, please add a short changelog entry.

File: internal/ui/restore/json.go [hunk 3]
```diff
@@ -108,6 +110,7 @@ type statusUpdate struct {
 	TotalFiles     uint64  `json:"total_files,omitempty"`
 	FilesRestored  uint64  `json:"files_restored,omitempty"`
 	FilesSkipped   uint64  `json:"files_skipped,omitempty"`
+	FilesDeleted   uint64  `json:"files_deleted,omitempty"`
```
Comment (MichaelEischer) [inline internal/ui/restore/json.go:113]: The new field must be documented at `doc/075_scripting.rst`.

Comment (konidev20) [reply internal/ui/restore/json.go:113]: I have updated the documentation with the `files_deleted` json field.

File: internal/restorer/restorer.go (hunk 2 above)
Comment (konidev20) [reply internal/restorer/restorer.go]: Done. 
1. I have removed the directory check. 
2. I have renamed the function to `ReportDeletion`

**Review by MichaelEischer — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**